xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/SaveAction.java

```diff
@@ -220,12 +224,20 @@ public boolean save(XWikiContext context) throws XWikiException
             tdoc.readFromForm(form, context);
         }
 
-        // TODO: handle Author
-        String username = context.getUser();
-        tdoc.setAuthor(username);
+        // TODO: entirely rely on UserReference from the context
```

Comment:
Going to be fun finding the right get* name with getUserReference already taken :)

Reply:
To be honest I started looking to perform the refactoring, and I quickly stopped given the nightmare it was between naming stuff,  use UserReference everywhere and initialize it properly where it needs...